src/raw/document_buf.rs

```diff
@@ -386,7 +386,7 @@ impl Deref for RawDocumentBuf {
 
 impl Borrow<RawDocument> for RawDocumentBuf {
     fn borrow(&self) -> &RawDocument {
-        &*self
+        &**self
```

Comment:
Why is the extra deref needed here?

Reply:
The error clippy made for this:
```
error: deref on an immutable reference
   --> src/raw/document_buf.rs:389:9
    |
389 |         &*self
    |         ^^^^^^
    |
    = note: `-D clippy::borrow-deref-ref` implied by `-D warnings`
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#borrow_deref_ref
help: if you would like to reborrow, try removing `&*`
    |
389 |         self
    |
help: if you would like to deref, try using `&**`
    |
389 |         &**self
    |
```

I've updated it to call `.deref()` directly for clarity.